src/amo/commands/updateTaskbar.ts

```diff
@@ -17,38 +20,44 @@ export async function updateTaskbar() {
     return;
   }
 
-  const path = doc.uri.fsPath;
+  const filePath = doc.uri.fsPath;
   const rootFolder = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
   if (!rootFolder) {
     return;
   }
-  const relativePath = path.replace(rootFolder, "");
-  const pathParts = relativePath.split("/");
-  let guid = pathParts[1];
-  const version = pathParts[2];
 
+  const relativePath = filePath.replace(rootFolder, "");
+  const [guid, version] = relativePath.split(path.sep).splice(-2);
   if (!guid || !version) {
     return;
   }
 
-  if (
-    guid.length === 36 &&
-    guid[8] === "-" &&
-    guid[13] === "-" &&
-    guid[18] === "-" &&
-    guid[23] === "-"
-  ) {
-    // looks like a guid
-    guid = "{" + guid + "}";
-  }
+  const reviewUrl = `${constants.reviewBaseURL}${guid}`;
+  try {
+    const response = await Promise.race([
+      fetch(reviewUrl),
+      new Promise<Response>((_, reject) =>
+        setTimeout(() => reject(new Error("Request timed out")), 2000)
+      ) as Promise<Response>,
+    ]);
 
-  const reviewUrl = `https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/${guid}`;
-  const response = await fetch(reviewUrl);
-  if (response.status === 404) {
+    if (response.status === 404) {
+      // not a review page
+      return;
+    } else if (response.status === 403) {
+      // not authed
+      return;
+    } else if (response.status !== 200) {
+      // other errors
+      return;
+    }
```

Comment:
seems like you can combine these all into the same `if` - especially as they're all `!== 200`.

Reply:
Also, maybe we should they raise an `Error` so it's caught and logged to the console like a timeout error would be.

Reply:
I had intended for 403 to prompt the user to sign in again and kept the if structure in case we wanted other errors to prompt the user too (i'm not sure what error codes you're working with). Since they do nothing now, I'll make non-200 raise an error as well. With caching, this part will disappear.